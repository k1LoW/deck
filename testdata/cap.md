# CAP theorem

## In Database theory

## Consistency

Every read receives the most recent write or an error.

## Availability

Every request received by a non-failing node in the system must result in a response.

## Partition tolerance

The system continues to operate despite an arbitrary number of messages being dropped (or delayed) by the network between nodes.

<!-- {"layout":"title-and-body-3col"} -->
