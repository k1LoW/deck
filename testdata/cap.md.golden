[
  {
    "layout": "title-and-body-3col",
    "titles": [
      "CAP theorem"
    ],
    "subtitles": [
      "In Database theory",
      "Consistency",
      "Availability",
      "Partition tolerance"
    ],
    "bodies": [
      {
        "paragraphs": [
          {
            "fragments": [
              {
                "value": "Every read receives the most recent write or an error."
              }
            ]
          }
        ]
      },
      {
        "paragraphs": [
          {
            "fragments": [
              {
                "value": "Every request received by a non-failing node in the system must result in a response."
              }
            ]
          }
        ]
      },
      {
        "paragraphs": [
          {
            "fragments": [
              {
                "value": "The system continues to operate despite an arbitrary number of messages being dropped (or delayed) by the network between nodes."
              }
            ]
          }
        ]
      }
    ]
  }
]